File: app/src/main/java/bassamalim/hidaya/models/AyatTelawaDB.java
package bassamalim.hidaya.models;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(tableName = "ayat_telawa", primaryKeys = {"rec_id", "rate"},
        foreignKeys = @ForeignKey(entity = AyatRecitersDB.class,
                parentColumns = "rec_id", childColumns = "rec_id"))
public class AyatTelawaDB {

    @ColumnInfo(name = "rec_id")
    private final int rec_id;
    @ColumnInfo(name = "rate")
    private final int rate;
    @ColumnInfo(name = "source")
    private final String source;

    public AyatTelawaDB(int rec_id, int rate, String source) {
        this.rec_id = rec_id;
        this.rate = rate;
        this.source = source;
    }

    public int getRec_id() {
        return rec_id;
    }

    public int getRate() {
        return rate;
    }

    public String getSource() {
        return source;
    }
}

File: app/src/main/java/bassamalim/hidaya/database/dbs/AyatTelawaDB.kt
package bassamalim.hidaya.database.dbs

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(
    tableName = "ayat_telawa",
    primaryKeys = ["rec_id", "rate"],
    foreignKeys = [ForeignKey(
        entity = AyatRecitersDB::class,
        parentColumns = arrayOf("rec_id"),
        childColumns = arrayOf("rec_id"),
        onUpdate = ForeignKey.CASCADE,
        onDelete = ForeignKey.SET_DEFAULT
    )]
)
class AyatTelawaDB(
    @field:ColumnInfo(name = "rec_id") private val recId: Int,
    @field:ColumnInfo(name = "rate") private val rate: Int,
    @field:ColumnInfo(name = "source") private val source: String?
) {
    fun getRecId(): Int {
        return recId
    }

    fun getRate(): Int {
        return rate
    }

    fun getSource(): String {
        return source!!
    }
}